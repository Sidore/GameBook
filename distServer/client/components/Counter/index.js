"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Counter = /** @class */ (function (_super) {
    __extends(Counter, _super);
    function Counter() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = {
            count: 0
        };
        _this.increment = function () {
            _this.setState({
                count: (_this.state.count + 1)
            });
        };
        _this.decrement = function () {
            _this.setState({
                count: (_this.state.count - 1)
            });
        };
        return _this;
    }
    Counter.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement("p", null, this.state.count),
            React.createElement("button", { onClick: this.increment }, "Increment"),
            React.createElement("button", { onClick: this.decrement }, "Decrement")));
    };
    return Counter;
}(React.Component));
exports.default = Counter;
