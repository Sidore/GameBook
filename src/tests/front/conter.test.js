import React from 'react';
import renderer from 'react-test-renderer';
import Counter from "../../client/components/Counter/index.tsx"

test('Counter', () => {
  const component = renderer.create(
    <Counter />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();


});