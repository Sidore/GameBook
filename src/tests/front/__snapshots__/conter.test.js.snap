// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter 1`] = `
<div>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
